src/main/java/de/bonndan/nivio/input/ItemRelationProcessor.java

```diff
@@ -66,29 +70,39 @@ public ProcessingChangelog process(@NonNull final LandscapeDescription input, @N
                     .filter(relation -> !processed.contains(relation))
                     .filter(relation -> origin.equals(relation.getSource()))
                     .forEach(relation -> {
-                        processLog.info(String.format(origin + ": Removing relation between %s and %s", relation.getSource(), relation.getTarget()));
-                        Item currentSource = landscape.getItems().pick(
-                                relation.getSource().getFullyQualifiedIdentifier().getItem(),
-                                relation.getSource().getFullyQualifiedIdentifier().getGroup()
-                        );
-                        if (!currentSource.removeRelation(relation)) {
-                            processLog.warn(String.format("Could not remove relation %s from source %s", relation, relation.getSource()));
-                        }
-
-                        Item currentTarget = landscape.getItems().pick(
-                                relation.getTarget().getFullyQualifiedIdentifier().getItem(),
-                                relation.getTarget().getFullyQualifiedIdentifier().getGroup()
-                        );
-                        if (!currentTarget.removeRelation(relation)) {
-                            processLog.warn(String.format("Could not remove relation %s from target %s", relation, relation.getSource()));
-                        }
+                        processLog.info(String.format("%s: Removing relation between %s and %s", origin, relation.getSource(), relation.getTarget()));
+                        removeRelationFromItem(landscape, relation, relation.getSource());
+                        removeRelationFromItem(landscape, relation, relation.getTarget());
                         changelog.addEntry(relation, ProcessingChangelog.ChangeType.DELETED, null);
                     });
         });
 
         return changelog;
     }
 
+    /**
+     * Gracefully finds the relation end item in the landscape and tries to remove the relation.
+     *
+     * @param landscape the current landscape
+     * @param relation the relation to remove
+     * @param relationEnd the relation source or target
+     */
+    private void removeRelationFromItem(Landscape landscape, Relation relation, Item relationEnd) {
```

Comment:
Wie wäre es wenn du direkt die Items injectest anstatt Landscape, denn ich habe mich zuerst gefragt warum Landscape gebraucht wird. Dann könnte die Methode auch removeRelationFromItems() heißen.

Reply:
Actually we pick one item only here, the whole collection is never accessed

Reply:
Sure it is in line 92 `landscape.getItems()`. 

If you have a signature like `private void removeRelationFromItem(List<Item> items, Relation relation, Item relationEnd)` the method call `removeRelationFromItem(landscape.getItems(), relation, relation.getSource());`would be more readable.

Reply:
`landscape.getItems()` returns an object, not a collection. We use an in-memory (CQEngine) here. I will consider rewriting the method.